### DISCUSSION TITLE
Supporting C#9 init-only properties for Commands

### DISCUSSION THREAD

**Discussion by zlavoie** [BrighterCommand/Brighter#1402] [Q&A]
**Is your feature request related to a problem? Please describe.**
C#9 came out with records and init-only properties. So now when trying to implement records, I am limited to using  'set' in IRequest. 

**Describe the solution you'd like**
 If the Id of IRequest can be init instead of set, then it can be used in a record and the comand can be fired to the handler.

**Describe alternatives you've considered**
I could use set, and then that is the only property that is NOT immutable, which starts to violate the purpose of a record in C#9.

**Comment by iancooper** [comment]
Yeah, we need to look at how we support C# 9 and even nullable reference types properly. One issue for us is backwards compatibility. For now, in a related example, it may make more sense for us to target LTS releases of .NET, so 6 vs 5. When looking at new language feature support i.e. treating a message as a record type, we need to think about breaking folks who are not using C# 9 yet. 

So it needs some thought to understand their experience

**Comment by iancooper** [comment]
@zlavoie Do you know anything about the backward compatibility of C# 9 init properties, I have not had a chance to look
